src/core/Akka/Actor/ActorCell.DeathWatch.cs

```diff
@@ -202,27 +207,23 @@ protected void RemWatcher(IActorRef watchee, IActorRef watcher)
 
         protected void AddressTerminated(Address address)
         {
-            var watchedBy = _state.GetWatchedBy();
             // cleanup watchedBy since we know they are dead
             MaintainAddressTerminatedSubscription(() =>
             {
-                foreach (var a in watchedBy.Where(a => a.Path.Address == address))
+                foreach (var a in _state.GetWatchedBy().Where(a => a.Path.Address == address))
                 {
                     //_watchedBy.Remove(a);
                     _state = _state.RemoveWatchedBy(a);
                 }
             });
 
-            //
-            watchedBy = _state.GetWatchedBy();
-
             // send DeathWatchNotification to self for all matching subjects
             // that are not child with existenceConfirmed = false because we could have been watching a
             // non-local ActorRef that had never resolved before the other node went down
             // When a parent is watching a child and it terminates due to AddressTerminated
             // it is removed by sending DeathWatchNotification with existenceConfirmed = true to support
             // immediate creation of child with same name.
-            foreach (var a in watchedBy.Where(a => a.Path.Address == address))
+            foreach (var a in _state.GetWatching().Where(a => a.Path.Address == address))
```

Comment:
So was this a bug that got added when we extracted the `ActorCell` state, or has this been a persistent issue for some time? Only asking because it may explain some of the cluster deathwatch behavior I've observed during multi-node test runs.